src/js/kits/model-kit-select/search-input.js

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+
+const SearchInput = (props) => {
+    const { onChange } = props;
+    const [query, setQuery] = useState('');
+    return (
+        <FormattedMessage
+            id="kits.filter.byName"
+            defaultMessage="filter by name">
+            {
+                placeholder =>
+                    <input
+                        type="text"
+                        className="form-control"
+                        value={query}
+                        onChange={(event) => {
+                            const newQuery = event.target.value;
+                            setQuery(newQuery);
+                            onChange(newQuery);
```

Comment:
Could maybe the query come directly from the props? Would remove the need to have duplicated state inside this component. 

Good reading on the matter: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
really don't like it in this case - this way the component is truly standalone. If I refactor that to have the query from the prop, I have my search params in the `ModelKitSelect`, which needs to pass the props down to `FilterForm`, which needs to pass it down to `SearchInput`. IMHO too many components get cluttered with props that aren't relevant for themselves.